test/pkg/environment/azure/expectations.go

```diff
@@ -0,0 +1,83 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+)
+
+func (env *Environment) EventuallyExpectKarpenterNicsToBeDeleted() {
+	GinkgoHelper()
+	Eventually(func() bool {
+		pager := env.InterfacesClient.NewListPager(env.NodeResourceGroup, nil)
+		for pager.More() {
+			resp, err := pager.NextPage(env.Context)
+			if err != nil {
+				return false
+			}
+
+			for _, nic := range resp.Value {
+				if nic.Tags != nil {
+					if _, exists := nic.Tags[strings.ReplaceAll(karpv1.NodePoolLabelKey, "/", "_")]; exists {
+						return false
+					}
+				}
+			}
+		}
+		return true
+	}).WithTimeout(10*time.Minute).WithPolling(10*time.Second).Should(BeTrue(), "Expected all orphan NICs to be deleted")
+}
+
+func (env *Environment) ExpectCreatedInterface(networkInterface armnetwork.Interface) {
+	GinkgoHelper()
+	poller, err := env.InterfacesClient.BeginCreateOrUpdate(env.Context, env.NodeResourceGroup, lo.FromPtr(networkInterface.Name), networkInterface, nil)
+	Expect(err).ToNot(HaveOccurred())
+	_, err = poller.PollUntilDone(env.Context, nil)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func (env *Environment) GetClusterSubnet() *armnetwork.Subnet {
+	GinkgoHelper()
+	vnet, err := getVNET(env.Context, env.VNETClient, env.VNETResourceGroup)
+	Expect(err).ToNot(HaveOccurred())
+	return vnet.Properties.Subnets[0]
+}
+
+func getVNET(ctx context.Context, client *armnetwork.VirtualNetworksClient, vnetRG string) (*armnetwork.VirtualNetwork, error) {
```

Comment:
This returns the first VNET found, right? Let's reflect it in the function name.
This (and any test using it) will fail in BYO VNET cluster, right? It is likely acceptable, but let's make a comment making it clear this is expected.

Reply:
Would it fail for BYO Vnet? If we pass in a proper custom vnet RG? Assuming that in test we just create a vnet and a single subnet for the cluster. 

Regardless we don't need to code for BYO vnet here.